code/modules/mob/living/carbon/skillchip.dm

```diff
@@ -0,0 +1,110 @@
+	// Grab the brain.
+	var/obj/item/organ/brain/brain = getorganslot(ORGAN_SLOT_BRAIN)
+
+	// Check for the brain. No brain = no implant.
+	if(QDELETED(brain))
+		return FALSE
+
+	// Check the chip can actually be implanted.
+	if(!force && skillchip.has_mob_incompatibility(src))
+		return FALSE
+
+	// Implant and call on_apply proc if successful.
+	if(brain.implant_skillchip(skillchip))
+		skillchip.on_implant(src, silent, TRUE)
+		skillchip.forceMove(brain)
+		return TRUE
+
+	return FALSE
+
+/**
+  * Attempts to remove this skillchip from the target carbon's brain.
+  *
+  * Returns FALSE when the skillchip couldn't be removed for some reason,
+  * including the target or brain not existing or the skillchip not being in the brain.
+  * Arguments:
+  * * target - The living carbon whose brain you want to remove the chip from.
+  * * silent - Whether or not to display the removal message.
+  */
+/mob/living/carbon/proc/remove_skillchip(obj/item/skillchip/skillchip, silent = FALSE)
+	// Check the target's brain, making sure the target exists and has a brain.
+	var/obj/item/organ/brain/brain = getorganslot(ORGAN_SLOT_BRAIN)
+	if(QDELETED(brain))
+		return FALSE
+
+	// Remove and call on_removal proc
+	if(!brain.remove_skillchip(skillchip))
+		stack_trace("Failed to remove skillchip [skillchip] from [src].")
+		return FALSE
+
+	skillchip.on_removal(src, silent)
+	return TRUE
+
+/**
+  * Creates a list of type paths of skillchips in the mob's brain.
+  *
+  * Returns a simple list of typepaths.
+  */
+/mob/living/carbon/proc/get_skillchip_type_list()
+	// Check the target's brain, making sure the target exists and has a brain.
+	var/obj/item/organ/brain/brain = getorganslot(ORGAN_SLOT_BRAIN)
+	if(QDELETED(brain))
+		return list()
+
+	return brain.get_skillchip_type_list()
+
+/**
+  * Destroys all skillchips in the brain, calling on_removal if the brain has an owner mob.
+  */
+/mob/living/carbon/proc/destroy_all_skillchips(silent = FALSE)
+	// Check the target's brain, making sure the target exists and has a brain.
+	var/obj/item/organ/brain/brain = getorganslot(ORGAN_SLOT_BRAIN)
+
+	if(QDELETED(brain))
+		return FALSE
+
+	brain.destroy_all_skillchips(silent)
+
+	return TRUE
+
+/// Disables or re-enables any extra skillchips after skillchip limit changes.
+/mob/living/carbon/proc/update_skillchips()
+	var/obj/item/organ/brain/brain = getorganslot(ORGAN_SLOT_BRAIN)
+
+	if(!brain)
+		return
+
+	var/limit = max_skillchip_slots
+	var/dt = limit - used_skillchip_slots
+	var/list/inactive_skillchips = list()
+
+	for(var/chip in brain.skillchips)
+		var/obj/item/skillchip/skillchip = chip
+		if(!skillchip.active)
+			inactive_skillchips += skillchip
+
+	// We have skillchips to deactivate
+	if(dt < 0)
+		//Might deactivate more than necessary but not worth sorting this
+		while(dt < 0)
+			var/obj/item/skillchip/skillchip = brain.skillchips[length(brain.skillchips)]
```

Comment:
Since this list now also contains inactive skillchips, you need to use something else here or you'll try to deactivate inactive ones.